dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/deduplication/DeduplicationController.java

```diff
@@ -229,9 +232,9 @@ private void validatePotentialDuplicate(PotentialDuplicate potentialDuplicate)
 
     checkValidTrackedEntity(potentialDuplicate.getDuplicate(), "duplicate");
 
-    canReadTrackedEntity(getTrackedEntity(potentialDuplicate.getOriginal()));
+    canReadTrackedEntity(trackedEntityService.getTrackedEntity(potentialDuplicate.getOriginal()));
```

Comment:
is the `canRead` check not already done by the service?

Reply:
This is true but the thrown exception will change once https://dhis2.atlassian.net/browse/DHIS2-18541 is completed.
When we are serving multiple and single endpoints with the same code we lose the ability to differentiate between 404 and 403.

Reply:
Is that just something to keep in mind, or are you saying we should continue returning 403 in this case?

Reply:
Just to keep in mind.
And maybe worth a discussion on the backend because we are declaring `ForbiddenException` in methods that cannot really throw such exception, and it is confusing.
(For example [here](https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/enrollment/DefaultEnrollmentService.java#L90-L107))